ships/0037-shipwright-build-strategy-controller.md

```diff
@@ -0,0 +1,190 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: shipwright-build-strategy-controller
+authors:
+  - "@jkhelil"
+reviewers:
+  - "@adambkaplan"
+  - "@qu1queee"
+approvers:
+  - "@adambkaplan"
+  - "@qu1queee"
+creation-date: 2023-10-02
+last-updated: 2023-10-02
+status: implementable
+---
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+
+## Summary
+
+This ship proposes support of installing a catalog of Build Strategies
+via the shipwright operator in order to improve the user experience around Shipwright Build
+
+## Motivation
+
+Actually with the Shipwright operator, the user installs the Shipwright Build Controller. 
+Installation and upgrade of Build Strategies and Cluster Strategies is currently a manual process
+In order to improve user experience around Shipwright, the operator should also install the build strategies
+
+### Goals
+
+- Create a Build Strategy catalog repository in the upstream shipwright-io organization
+- Copy/publish the Build strategies from the upstream build repository to the new create catalog repository
+- Allow a user to install the default Build Strategy Catalog or if applicable the user provided catalog, via the Shipwright Operator
+ 
+### Non-Goals
+
+- Add a new operator to install the Build Strategy catalog
+- Curate or rework the Build Strategies in the upstream build repository when copied to the new catalog repository
+
+## Proposal
+
+We propose to introduce a new CRD in the Shipwright Operator : `ShipwrightBuildStrategy` API.
+
+```yaml
+apiVersion: operator.shipwright.io/v1alpha1
+kind: ShipwrightBuildStrategy
```

Comment:
Do we really need a new CRD? Can't we extend the existing `ShipwrightBuild` CRD instead?

Reply:
Initially I was thinking about avoiding concurrency issues when updating the CR status, but now that you raise the comment and after thinking about it, We do not need a new CRD, as the controller design is leveraging the optimistic concurrency and if the update fails, we should requeue
I will update the SHIP using the same CRD ShipwrightBuild